speechbrain/core.py

```diff
@@ -882,9 +883,14 @@ def fit_batch(self, batch):
                     self.optimizer.step()
                 self.optimizer.zero_grad()
                 self.optimizer_step += 1
-
+        if should_step:
+            self.on_fit_batch_end()
         return loss.detach().cpu()
 
+    def on_fit_batch_end(self):
+        """Called after ``fit_batch()``"""
+        pass
+
```

Comment:
I like this hook, since it makes it less likely people will have to override `fit_batch`, which has grown somewhat in complexity since we first implemented it. Would it make sense to have a similar function at the beginning, `on_fit_batch_start`?

Reply:
I guess so, although I imagine the point of `on_fit_batch_start` is for people to do stuff with results like the loss etc. that have been calculated, what would `on_fit_batch_start` take as input?

Reply:
If there's no obvious use case for `on_fit_batch_start` then I guess we should wait to add it until its needed under the YAGNI principle.

Reply:
Do we need any significant effort to migrate metric computation code to the on_fit_batch_end?

Reply:
I don't think so, since AFAIK there is no metric computation stuff in core.py ?

Reply:
I guess it doesn't need to be part of this PR, but I believe we should start another once this is in to migrate the `templates` at the very least to do metric computation in this function, as that would be much cleaner.